superset/sql_parse.py

```diff
@@ -128,3 +134,43 @@ def __extract_from_token(self, token):
                 for token in item.tokens:
                     if self.__is_identifier(token):
                         self.__process_identifier(token)
+
+    def _get_limit_from_token(self, token):
+        if token.ttype == sqlparse.tokens.Literal.Number.Integer:
+            return int(token.value)
+        elif token.is_group:
+            return int(token.get_token_at_offset(1).value)
+
+    def _extract_limit_from_outermost_layer(self, statement):
+        limit_token = None
+        for pos, item in enumerate(statement.tokens):
+            if item.ttype in Keyword and item.value.lower() == 'limit':
+                limit_token = statement.tokens[pos + 2]
+                break
+        if not limit_token:
+            return limit_token
+        return self._get_limit_from_token(limit_token)
+
+    def get_query_with_new_limit(self, new_limit):
+        """returns the query with the specified limit"""
+        """does not change the underlying query"""
+        if not self._limit:
+            return self.sql + ' LIMIT ' + str(new_limit)
+        limit_pos = None
+        # Add all items to before_str until there is a limit
+        for pos, item in enumerate(self._parsed[0].tokens):
+            if item.ttype in Keyword and item.value.lower() == 'limit':
+                limit_pos = pos
+                break
+        limit = self._parsed[0].tokens[limit_pos + 2]
+        if limit.ttype == sqlparse.tokens.Literal.Number.Integer:
+            self._parsed[0].tokens[limit_pos + 2].value = new_limit
```

Comment:
The `self._parsed[0].tokens` array contains a list of references to the `Token` objects and thus you can mutate the token in-place (the reference remains unchanged) so this can simply be 
```
limit.value = new_limit
```